### PR TITLE
test: Add basic test cases for reconciliation

### PR DESCRIPTION
**Pull request by builditben**
This PR adds some test scenarios for the reconciliation API

### CODE REVIEW

**Review by DafyddLlyr — APPROVED** [review]
Thanks very much for picking this up 👍 

Just one small question - not a blocker!

File: api.planx.uk/saveAndReturn/validateSession.ts [hunk 1]
```diff
@@ -164,41 +168,46 @@ const updateLowcalSessionData = async (
         pk_columns: { id: $sessionId }
         _set: { data: $data }
       ) {
-        id
         data
-        created_at
-        updated_at
```
Comment (builditben) [inline api.planx.uk/saveAndReturn/validateSession.ts:170]: These fields aren't used so I've tightened up the query here.

File: api.planx.uk/saveAndReturn/validateSession.ts [hunk 2]
```diff
@@ -134,22 +138,32 @@ const validateSession = async (
...
   const query = gql`
-    query FindSession {
-      lowcal_sessions {
+    query FindSession($sessionId: uuid!, $email: string!) {
+      lowcal_sessions(
+        where: {
+          sessionId: { _eq: $sessionId }
+          email: { _eq: $email }
+          limit: 1
+        }
```
[the remark this answers is not in the document]
Comment (DafyddLlyr) [reply api.planx.uk/saveAndReturn/validateSession.ts]: Yep perfect, just wanted to be sure I wasn't missing anything 👍

[the remark this answers is not in the document]
Comment (builditben) [reply api.planx.uk/saveAndReturn/validateSession.ts]: I did have an error in my query that e2e tests caught so I've updated this now (still to be more explicit).

File: api.planx.uk/saveAndReturn/validateSession.test.ts [hunk 3]
```diff
@@ -1,25 +1,172 @@
...
+      .send(data)
+      .expect(200)
+      .then((response) => {
+        expect(response.body).toHaveProperty("alteredNodes", null);
+        expect(response.body).toHaveProperty("message", expectedMessage);
+        expect(response.body).toHaveProperty("removedBreadcrumbs", null);
+      });
+  });
+
+  it("shows a diff for changed nodes", async () => {
```
Comment (jessicamcinchak) [inline api.planx.uk/saveAndReturn/validateSession.test.ts:88]: nit: i'd find these positive test descriptions slightly more helpful if they referenced the specific response property they're testing

eg - this one is maybe more like `it("populates alteredNodes with the diff of changed nodes",...`

and then maybe `it.todo("populates removedBreadcrumbs when alteredNodes intersects with the session breadcrumbs")`? (happy for this case to be addressed later!)

**Review by jessicamcinchak — APPROVED** [review]
thanks for these tests & mocks!

**Pull request merged**